app/src/main/java/org/mozilla/fenix/components/metrics/AppLaunchTimeMeasurement.kt

```diff
@@ -0,0 +1,51 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.metrics
+
+import android.os.Process
+import android.os.SystemClock
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.launch
+import org.mozilla.fenix.perf.Stat
+
+/**
+ * Handles the logic of figuring out launch time for cold, warm and hot startup.
+ */
+class AppLaunchTimeMeasurement(private val stats: Stat = Stat()) {
+
+    // when user launched the time. i.e press app_icon, launched a link.
+    private var processLaunchTimeStampNanoSeconds: Long? = null
```

Comment:
It's kind of confusing that we're re-using this variable to mean many things:
- processLaunch time for COLD
- activity init time for WARM
- onRestart time for HOT

Why don't we just set them as different variables? Before reporting a value, we then have the ability to validate that only one value is set (as we'd expect) and report an error if not. This would make the code a lot more explicit: right now it's very unclear which lifecycle method refers to which startup case and why we set this value so many times.

Reply:
I changed the code to align more with this